### PR TITLE
Login guarded routes

### PR DESCRIPTION
**Pull request by tomheller**
Todo: 
* [x] Move Menu into a better structure to support more screens
* [x] Add enter/leave animation for header bar to make the transition nicer

### CODE REVIEW

**Comment by tomheller** [comment]
Open for review before this is running too long. 
## Done: 
* guarded login route 
* Refactored menu to be more extensible
* Added initial settings page for jira integration
* Added animation for login

File: src/router/index.js [hunk 1]
```diff
@@ -2,9 +2,20 @@ import Vue from 'vue';
 import Router from 'vue-router';
 import AppMain from '@/components/AppMain';
 import AppHeader from '@/components/AppHeader';
+import AppLogin from '@/components/AppLogin';
+import Settings from '@/components/Settings';
+import Service from '@/service';
 
 Vue.use(Router);
 
+const checkLoggedin = (to, from, next) => {
```
Comment (ffriedl89) [inline src/router/index.js]: camelCase => checkLoggedIn ?

**Review by ffriedl89 — APPROVED** [review]
looks good, i haven't seen the animations yet, but pretty nice that we have some already 🌵

**Comment by tomheller** [comment]
Animations: 
nov-02-2017 21-42-22

**Comment by ffriedl89** [comment]
🍔  nice!

**Pull request merged**